mitodl/mit-open
# adjust / refactor channel detail header

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4781

### Description (What does it do?)
This PR refactors `ChannelPageSkeleton.tsx` into two separate files; `DefaultChannelSkeleton.tsx` and `UnitChannelSkeleton.tsx`. The reason for this is that the layout of the `Banner` sections of the Unit channels is different enough that it was starting to become difficult to make changes to the Unit specific layout without affecting the other types of channels. This defines a clear separation so that Unit channels that have more extensive branding / styling can be customized without worrying about affecting the other types of channel pages. The use of `BannerPage` was replaced with simply using the newer `Banner` component. Slots were added to `Banner` to allow customization in the way the Unit channel pages need. Lastly the original problem noted in the issue above was addressed, which was just that the non-Unit channel page banners looked off from the design.

### Screenshots (if appropriate):
![image](https://github.com/mitodl/mit-open/assets/12089658/c5689025-c2f8-4b13-ae2d-b4e9680fb5b7)
![image](https://github.com/mitodl/mit-open/assets/12089658/1f392618-e68e-4a87-a14d-86706688d3b2)
![image](https://github.com/mitodl/mit-open/assets/12089658/b6f60446-7eac-4464-b1d6-b527a0c62a6d)
![image](https://github.com/mitodl/mit-open/assets/12089658/18e5d9cc-89d8-46a6-9eb8-2c371f79265f)
![image](https://github.com/mitodl/mit-open/assets/12089658/a47ee900-bd4a-452a-86d8-8d43e56215df)
![image](https://github.com/mitodl/mit-open/assets/12089658/519ae958-d724-477d-90af-90cda9ea2ed3)


### How can this be tested?
 - Spin up `mit-open` on this branch
 - Ensure your departments, topics and units are populated
   - `docker compose exec web ./manage.py update_offered_by`
   - `docker compose exec web ./manage.py update_topics`
   - `docker compose exec web ./manage.py update_departments_schools`
 - Visit the home page at http://localhost:8062/
 - Click the icon to open the nav menu, then click "By Topic"
 - Select any of the topics and verify that the text in the banner does not shift and lines up with the listing page
 - Click the icon to open the menu again and click "By Provider"
 - Click any of the units on the listing page
 - Verify that the layout of the unit channel pages has not changed
 - Repeat this process on mobile from the home page step
